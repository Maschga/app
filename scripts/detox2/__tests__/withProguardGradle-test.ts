import { addDetoxProguardRules } from "../withProguardGradle";

const buildGradleFixture = `

android {
    compileSdkVersion rootProject.ext.compileSdkVersion
    
    buildTypes {
        debug {
            signingConfig signingConfigs.debug
        }
        release {
            // Caution! In production, you need to generate your own keystore file.
            // see https://reactnative.dev/docs/signed-apk-android.
            signingConfig signingConfigs.debug
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
        }
    }
}
`;
const buildGradleWithoutProguardFixture = `

android {
    compileSdkVersion rootProject.ext.compileSdkVersion
    
    buildTypes {
        debug {
            signingConfig signingConfigs.debug
        }
        release {
            // Caution! In production, you need to generate your own keystore file.
            // see https://reactnative.dev/docs/signed-apk-android.
            signingConfig signingConfigs.debug
        }
    }
}
`;

describe(addDetoxProguardRules, () => {
  it(`adds proguard rules to gradle`, () => {
    const applied = addDetoxProguardRules(buildGradleFixture);
    expect(applied).toMatchSnapshot();
    expect(applied).toMatch(/detox/);
    expect(applied).toMatch(/proguardFile\(detoxProguardRulesPath\)/);
  });
  it(`doesn't add proguard rules to gradle if proguard is missing from the template`, () => {
    expect(
      addDetoxProguardRules(buildGradleWithoutProguardFixture),
    ).not.toMatch(/detox/);
  });
  it(`doesn't duplicate`, () => {
    const applied = addDetoxProguardRules(buildGradleFixture);
    expect(addDetoxProguardRules(applied)).toStrictEqual(applied);
  });
});
