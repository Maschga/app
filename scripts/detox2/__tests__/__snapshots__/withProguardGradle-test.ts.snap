// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addDetoxProguardRules adds proguard rules to gradle 1`] = `
"

android {
    compileSdkVersion rootProject.ext.compileSdkVersion
    
    buildTypes {
        debug {
            signingConfig signingConfigs.debug
        }
        release {
            // Caution! In production, you need to generate your own keystore file.
            // see https://reactnative.dev/docs/signed-apk-android.
            signingConfig signingConfigs.debug
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
            // Detox-specific additions to pro-guard
            def detoxProguardRulesPath = new File(["node", "--print", "require.resolve('detox/package.json')"].execute(null, rootDir).text.trim(), "../android/detox/proguard-rules-app.pro")
            proguardFile(detoxProguardRulesPath)
            
        }
    }
}
"
`;
